rust/lance-index/src/vector/graph/memory.rs

```diff
@@ -33,6 +35,26 @@ impl InMemoryVectorStorage {
 
 impl VectorStore for InMemoryVectorStorage {
     type DistanceCalculator<'a> = InMemoryDistanceCal;
+
+    fn try_from_batch(batch: &RecordBatch, distance_type: DistanceType) -> Result<Self>
+    where
+        Self: Sized,
+    {
+        let vectors = batch
+            .column_by_name("flat")
+            .ok_or(Error::Schema {
+                message: format!("column flat not found"),
+                location: location!(),
+            })?
+            .as_fixed_size_list();
+        let mat = MatrixView::<Float32Type>::try_from(vectors)?;
+        Ok(Self::new(mat, distance_type))
+    }
+
+    fn to_batch(&self) -> Result<RecordBatch> {
+        unimplemented!()
```

Comment:
why is this unimplemented?

Another nit would be, maybe we should rename this struct to `RawVectorStorage`?

Reply:
going to change this to store `FixedSizeList` and rename it to `FlatVectorStorage`

Reply:
still working on it, going to change the fields to `FixedSizeList` so it can support any types of vector, and rename it to `FlatVectorStorage`